src/frontend/type_checker.cpp

```diff
@@ -269,9 +293,17 @@ void TypeChecker::visit(FuncDecl::Ptr decl) {
       reportMultipleDefs("function parameter", name, genericParam);
     }
 
-    const Type::Ptr type = (genericParam->type == GenericParam::Type::RANGE) ?
-                           to<Type>(makeTensorType(ScalarType::Type::INT)) :
-                           to<Type>(env.getSetDefinition(genericParam));
+    Type::Ptr type;
+    if (genericParam->type == GenericParam::Type::RANGE) {
+      type = to<Type>(makeTensorType(ScalarType::Type::INT));
+    }
+    else {
+      SetType::Ptr setDef = env.getSetDefinition(genericParam);
+      // We have to guard for undefined nullptr set definitions.
+      if (setDef) {
```

Comment:
You're right, this was handling an older version of how I was attempting to handle making SetType an abstract class with subclasses. I think my current iteration (with SetType::getUndefinedSetType) is neater, but let me know if there's a better design pattern.